### PR TITLE
Fix anticheat resetting client position after the client is teleported.

### LINKED ISSUE #5118: resetting position after setpos
Most times a player is teleported, the anti-cheat resets their position.
This seems to be a major problem with the anti-cheat system.

### PR DESCRIPTION
**Pull request by Ekdohibs**
Previously, m_move_pool could accomodate the client moving from the new
position to the old one, and the server accepted the client to go back
to its old position. However, it couldn't then accomodate the client
moving from its old to its new position, and therefore would reset position
to the old position. Thus, by emptying m_move_pool after a teleport, the
server no longer accepts the client to go back to its old position. A
drawback is however that a laggy client *will* trigger a few
"moved_too_fast" anticheats before being told about its new position.

Fixes #5118.

To test, add a connexion delay to localhost (see https://daniel.haxx.se/blog/2010/12/14/add-latency-to-localhost/ for example).

### CODE REVIEW

File: src/content_sao.h [hunk 1]
```diff
@@ -169,6 +169,10 @@ class LagPool
 		if(m_pool < 0)
 			m_pool = 0;
 	}
+	void empty()
+	{
+		m_pool = m_max;
+	}
```
Comment (nerzhul) [inline src/content_sao.h]: missing empty line before and after function

Comment (Ekdohibs) [inline src/content_sao.h]: done

**Comment by nerzhul** [comment]
doesn't build on windows

**Comment by Ekdohibs** [comment]
@nerzhul that's just the build unreliably failing: `Connecting to minetest.kitsunemimi.pw (minetest.kitsunemimi.pw)|2a01:6f0:ffff:da::6b49|:80... failed: Network is unreachable.`

**Comment by nerzhul** [comment]
@sfan5 ?

**Comment by sfan5** [comment]
Network issues on travis backend (google cloud)

**Comment by sofar** [comment]
:+1:

**Comment by paramat** [comment]
WIP?

**Comment by sfan5** [comment]
doesn't look wip to me

**Comment by paramat** [comment]
Nore may be considering a change http://irc.minetest.net/minetest-dev/2017-01-30#i_4800556

**Comment by Ekdohibs** [comment]
Yes, I'll be fixing that as well I think, we shouldn't report that the player cheated in such cases. Expect the updated PR coming in less than a few hours.

**Comment by paramat** [comment]
d873545ac70331a224967493f9296a854d292dd8

**Pull request closed without merging**